service/controller/unittest/default_k8sclient.go

```diff
@@ -0,0 +1,82 @@
+package unittest
+
+import (
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/v2/pkg/apis/infrastructure/v1alpha2"
+	releasev1alpha1 "github.com/giantswarm/apiextensions/v2/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/apiextensions/v2/pkg/clientset/versioned"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8scrdclient"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	fakek8s "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+type fakeK8sClient struct {
+	ctrlClient client.Client
+	k8sClient  *fakek8s.Clientset
+}
+
+func FakeK8sClient() k8sclient.Interface {
```

Comment:
You might be able to use https://github.com/giantswarm/k8sclient/blob/master/pkg/k8sclienttest/empty.go for this. Not sure about the scheme part.

Reply:
Yep looks like it -- thanks. I hadn't seen that used anywhere. The full fake client might be necessary to test the cluster listing but I didn't plan to do that right now